kotlinx-coroutines-core/common/src/flow/operators/Delay.kt

```diff
@@ -40,36 +40,44 @@ import kotlin.time.*
  */
 @FlowPreview
 public fun <T> Flow<T>.debounce(timeoutMillis: Long): Flow<T> {
-    require(timeoutMillis > 0) { "Debounce timeout should be positive" }
-    return scopedFlow { downstream ->
-        // Actually Any, KT-30796
-        val values = produce<Any?>(capacity = Channel.CONFLATED) {
-            collect { value -> send(value ?: NULL) }
-        }
-        var lastValue: Any? = null
-        while (lastValue !== DONE) {
-            select<Unit> {
-                // Should be receiveOrClosed when boxing issues are fixed
-                values.onReceiveOrNull {
-                    if (it == null) {
-                        if (lastValue != null) downstream.emit(NULL.unbox(lastValue))
-                        lastValue = DONE
-                    } else {
-                        lastValue = it
-                    }
-                }
+    if (timeoutMillis <= 0) return this
```

Comment:
Can you please elaborate on the reasoning behind the change from `require(timeoutMillis > 0)` to `if (timeoutMillis <= 0) return this`? Why would you need to silently ignore zero or negative values here?

Reply:
I interpreted 0 `timeoutMillis` as no debounce and emit right away. As a user of this API, if you passed a 0 for whatever reason, I think I would expect it to just not debounce.

Also, in this variation `debounce(timeoutMillisSelector: (T) -> Long)`, the intention was allow 0ms to skip a debounce in certain conditions. It would match the behavior between the two variations.

There is a relevant comment in [here](https://github.com/Kotlin/kotlinx.coroutines/pull/2148#discussion_r457106584).